src/particles/elements/DipEdge.H

```diff
@@ -0,0 +1,88 @@
+/* Copyright 2021 Chad Mitchell, Axel Huebl
+ *
+ * This file is part of ImpactX.
+ *
+ * License: BSD-3-Clause-LBNL
+ */
+#ifndef IMPACTX_DIPEDGE_H
+#define IMPACTX_DIPEDGE_H
+
+#include "particles/ImpactXParticleContainer.H"
+
+#include <AMReX_Extension.H>
+#include <AMReX_REAL.H>
+
+#include <cmath>
+
+
+namespace impactx
+{
+    struct DipEdge
+    {
+        using PType = ImpactXParticleContainer::ParticleType;
+
+        /** Edge focusing associated with bend entry or exit
+         *
+         * @param psi Pole face angle in rad.
+         * @param rc Radius of curvature in m.
+         * @param g Gap parameter in m.
+         * @param K2 Fringe field integral (unitless).
+         */
+        DipEdge( amrex::ParticleReal const psi, amrex::ParticleReal const rc, amrex::ParticleReal const
+        g, amrex::ParticleReal const K2)
+        : m_psi(psi), m_rc(rc), m_g(g), m_K2(K2)
+        {
+        }
+
+        /** This is a dipedge functor, so that a variable of this type can be used like a
+         *  dipedge function.
+         *
+         * @param p Particle AoS data for positions and cpu/id
+         * @param px particle momentum in x
+         * @param py particle momentum in y
+         * @param pt particle momentum in t
+         * @param refpart reference particle
```

Comment:
```suggestion
         * @param refpart reference particle (unused)
```

Reply:
This is curious: why don't we need the reference particle for this implementation when we do need it for the simple sbend?

Reply:
This model of the edge focusing depends only on the rotation angle of the magnet pole face and the magnet radius of curvature; it does not depend on the reference energy.  However, other more sophisticated models of the pole face rotation and the dipole fringe field could require the reference values.  That is why I left this as an input argument.